packages/menhir/menhir.20200121/opam

```diff
@@ -0,0 +1,28 @@
+opam-version: "2.0"
+maintainer: "francois.pottier@inria.fr"
+authors: [
+  "François Pottier <francois.pottier@inria.fr>"
+  "Yann Régis-Gianas <yrg@pps.univ-paris-diderot.fr>"
+]
+homepage: "http://gitlab.inria.fr/fpottier/menhir"
+dev-repo: "git+https://gitlab.inria.fr/fpottier/menhir.git"
+bug-reports: "menhir@inria.fr"
+build: [
+  ["dune" "build" "@install" "-j" jobs]
+]
+install: [
+  ["dune" "install" "-j" jobs]
+]
```

Comment:
dune install is a command only necessary when installing from source without opam. Opam already takes care of the installation. This field can be removed.

Reply:
Ah nevermind I see you're installing several packages at once. Mmh however this feature uses ocamlfind, so ocamlfind should be added to the dependencies (see https://github.com/ocaml/dune/issues/3025)

Reply:
Another issue is that this command will also install some files in `<prefix>/lib/coq-menhirlib` even if coq is not installed or the library not built (an empty META/dune-package file is installed at the very least)

Reply:
Regarding the `install` field in the opam file: it is indeed needed (I tried removing it, and `dune` installed nothing at all).

Regarding `ocamlfind`: I would rather avoid a dependency on `ocamlfind` if possible. Are you certain that installing several packages at once requires `ocamlfind`? (Where is this documented?) (Why should installing several packages at once be more complicated than installing one package at a time?)

Reply:
Regarding your remark that "this command will also install some files in `<prefix>/lib/coq-menhirlib`": I think I just need the remove the declaration `(package (name coq-menhirlib))` in the file `dune-project`, right?

Reply:
Following up on myself: if I remove the declaration `(package (name coq-menhirlib))` then `dune` complains about the presence of the file `coq-menhirlib.opam` when I try to build Menhir.

I do not see clearly how I should declare that I have both ocaml/opam packages and coq/opam packages in the same project tree...

Reply:
It's not that it is required for several packages, it is more like it is required when using `dune install` because dune install uses ocamlfind when available (e.g. https://github.com/ocaml/dune/blob/master/src/dune/context.ml#L247).

https://github.com/ocaml/dune/issues/3025#issuecomment-574149138 gives an answer from the dune team. ocamlfind is required to avoid crashes during installation in a local switch when ocamlfind is installed globally, so to avoid surprises for users it is best to install ocamlfind in every case.

------

One other possible fix would be to stop using dune install all together and instead split the project into 3, install them normally using opam and require them in the menhir package. Dune is meant to be used this way, using dune install is more of a "hack" when used in opam.

I can probably manage to provide a patch if that's a decent solution for you.

Reply:
Posted in https://gitlab.inria.fr/fpottier/menhir/issues/32

Reply:
> One other possible fix would be to stop using dune install all together and instead split the project into 3, install them normally using opam

Do you still recommend that I add a dependency on `ocamlfind`?

Reply:
with my patch there is no need for `dune install` thus no need for `ocamlfind`